network/fuzz.go

```diff
@@ -0,0 +1,46 @@
+// +build gofuzz
+package network
+
+import (
+	"bytes"
+	"fmt"
+)
+
+func Fuzz(data []byte) int {
```

Comment:
The return value appears to be a zero for failure and non-zero for success. How about returning an `error` on failure and `nil` on success instead?